lib/drops/type.ex

```diff
@@ -246,10 +246,16 @@ defmodule Drops.Type do
         )
       end
 
-      def new(spec, constraints) when is_list(constraints) do
-        new(
-          primitive: infer_primitive(spec),
-          constraints: infer_constraints({:type, {spec, constraints}})
+      def new(predicates, opts) when is_list(opts) do
+        type = new(opts)
+        constraints =
+          case type.constraints do
+            {:and, constraints} -> {:and, constraints ++ infer_constraints(predicates)}
```

Comment:
I made the minimal changes to make this work.  But it looks like `{:and, constraints}` gets treated the same as just `constraints`, so I wonder if we can just get rid of `{:and, ...}` everywhere.